src/audio/pcm_converter/pcm_converter_generic.c

```diff
@@ -136,6 +138,63 @@ static void pcm_convert_s32_to_s24(const struct audio_stream *source,
 
 #endif /* CONFIG_FORMAT_S24LE && CONFIG_FORMAT_S32LE */
 
+#if CONFIG_FORMAT_FLOAT && (CONFIG_FORMAT_S16LE || CONFIG_FORMAT_S24LE || CONFIG_FORMAT_S32LE)
+
+static inline int32_t _pcm_shift(int32_t d, int32_t a)
+{
+	int64_t dd = d;
+
+	if (a > 32)
+		a = 32;
+	else if (a < -32)
+		a = -32;
+
+	dd = a >= 0 ? dd << a : dd >> -a;
+	if (dd > INT32_MAX)
+		dd = INT32_MAX;
+
+	return (int32_t)dd;
+}
+
+#define PCM_ABS32(x) (((x) ^ ((x) >> 31)) + (int)((unsigned int)(x) >> 31))
+
+static int32_t _pcm_convert_f_to_i(int32_t src)
```

Comment:
inline comment, Btw if we are converting float to int then why do we pass in an int ?

Reply:
to don't provoke software float arithmetic library include to SOF FW for such a pipeline:
(fixed-s32) -> converter -> (float) -> external processing component -> (float) -> converter -> (fixed-s32)

Reply:
Are you meaning top exclude some casting support functions from the C  library or from compiler builtins ? If so, this needs to be made in comments next to each conversion function with reasons why.

Reply:
compiler builtins, updated comment